activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -228,14 +237,16 @@ def setup
 
     def test_structure_load
       filename = "awesome-file.sql"
-      Kernel.expects(:system).with("psql -q -f #{filename} my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', filename, 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
 
       ActiveRecord::Tasks::DatabaseTasks.structure_load(@configuration, filename)
     end
 
     def test_structure_load_accepts_path_with_spaces
       filename = "awesome file.sql"
-      Kernel.expects(:system).with("psql -q -f awesome\\ file.sql my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', 'awesome\\ file.sql', 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
```

Comment:
Just realized that while that test passes, in the real world the code would fail. That is, can't use filenames with escaped spaces with the current implementation of run_cmd ( btw I don't like that name ).  Aside from performance, wouldn't it be feasible to re-write tests to use less mocking and test the real deal? You'll have to excuse me if I'm a bit naive on these issues.